scripts/globals/weaponskills.lua

```diff
@@ -845,41 +849,44 @@ end;
 
 function getMultiAttacks(attacker, numHits)
     local bonusHits = 0;
-    local tripleChances = 1;
+    local multiChances = 1;
     local doubleRate = attacker:getMod(MOD_DOUBLE_ATTACK)/100;
     local tripleRate = attacker:getMod(MOD_TRIPLE_ATTACK)/100;
+    local quadRate = attacker:getMod(MOD_QUAD_ATTACK)/100;
+    
+    if (attacker:isPC()) then -- some mobskills use this function so we have to check if this is a player.
+        doubleRate = doubleRate + attacker:getMerit(MERIT_DOUBLE_ATTACK_RATE)/100;
+        tripleRate = tripleRate + attacker:getMerit(MERIT_TRIPLE_ATTACK_RATE)/100;
+    end
 
-    -- triple only procs on first hit, or first two hits if dual wielding
-    if (attacker:getOffhandDmg() > 0) then
-        tripleChances = 2;
+    -- QA/TA/DA can only proc on the first hit of each weapon or each fist
+    if (attacker:getOffhandDmg() > 0 or attacker:getWeaponSkillType(SLOT_MAIN) == SKILL_H2H) then
+        multiChances = 2;
     end
 
-    for i = 1, numHits, 1 do
-        chance = math.random();
-        if (chance < tripleRate and i <= tripleChances) then
+    for i = 1, multiChances, 1 do
+        if (math.random() < quadRate) then
```

Comment:
you should only need one math.random - if its between 0 and QA, its a quad, if its between QA and TA + QA, its a TA, if its between TA + QA and QA + TA + DA its a DA, otherwise its a single